updatepermissionbuilder.go

```diff
@@ -24,30 +28,45 @@ func (upb *UpdatePermissionBuilder) UpdatePermission(address, channel *xdr.ID, n
 	return upb
 }
 
-func (upb *UpdatePermissionBuilder) Action(permissionAction int32) *UpdatePermissionBuilder {
-	upb.permissionAction = xdr.PermissionAction(permissionAction)
+func (upb *UpdatePermissionBuilder) Address(address xdr.ID) *UpdatePermissionBuilder {
+	upb.key = address
 	return upb
 }
 
-func (upb *UpdatePermissionBuilder) Address(address xdr.ID) *UpdatePermissionBuilder {
-	upb.key = address
+func (upb *UpdatePermissionBuilder) Alias(alias string) *UpdatePermissionBuilder {
+	upb.alias = alias
+	return upb
+}
+
+func (upb *UpdatePermissionBuilder) Authorize(accountUpdateType xdr.AccountUpdateType, account xdr.ID, alias string, authorize bool) *UpdatePermissionBuilder {
+	upb.accountUpdateType = accountUpdateType
+	upb.authorizedAccount = account
+	upb.alias = alias
+	upb.authorize = authorize
 	return upb
 }
 
 func (upb *UpdatePermissionBuilder) Sign(pk ed25519.PrivateKey) (*xdr.Transaction, error) {
 
 	action := xdr.Action{
-		Address:   *upb.address,
-		ChannelID: *upb.channel,
-		Nonce:     upb.nonce,
-        BlockExpirationNumber: upb.blockExpirationNumber,
+		Address:               *upb.address,
+		ChannelID:             *upb.channel,
+		Nonce:                 upb.nonce,
+		BlockExpirationNumber: upb.blockExpirationNumber,
 		Category: xdr.ActionCategory{
 			Type: xdr.ActionCategoryTypeUPDATE,
 			Update: &xdr.Update{
-				Type: xdr.UpdateTypePERMISSION,
-				Permission: &xdr.Permission{
-					Action: upb.permissionAction,
-					Key:    upb.key,
+				Type: xdr.UpdateTypeACCOUNT,
+				Account: &xdr.AccountUpdate{
+					Type:  upb.accountUpdateType,
+					Alias: &upb.alias,
```

Comment:
The Alias field should not be set here if this builder is only for Authorization update transactions.

Reply:
Done. Thanks.